Added Viking Combat martial art

Hymore246 opened the pull request:
<!--
### How to use
Leave the headings unless they don't apply to your PR, replace commented out text ( surrounded with <!-- and -​-> ) with text describing your PR.
-->

#### Summary
<!--
A one-line description of your change that will be extracted and added to the project changelog at https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt
The format is: ```SUMMARY: Category "description"```  
The categories to choose from are: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N  
Example: ```SUMMARY: Content "Adds new mutation category 'Mouse'"```
See the Changelog Guidelines at https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md for explanations of the categories.
-->

`SUMMARY: Feature "Added new Martial Art: Viking Combat"`
#### Purpose of change
<!--
If there's an existing issue describing the problem this PR addresses or the feature it adds, please link it like: ```#1234```  
If it *fully* resolves an issue, link it like: ```Fixes #1234```  
Even if the issue describes the problem, please provide a few-sentence summary here.  
Example: ```Fixes #1234 - XL mutants cannot wear arm/leg splints due to missing OVERSIZE flag.```  
If there is no related issue, please describe the issue you are addressing, including how to trigger a bug if this is a bugfix.
-->

Adds Viking Combat as a new melee weapon martial art to the game. 
#### Describe the solution
<!--
How does the feature work, or how does this fix a bug?  
The easier you make your solution to understand, the faster it can get merged.
-->

Viking Combat is martial art based on the historical Viking combat tactics. Vikings were know it use surprise attacks and hit-and-run tactics against enemies. They were also very violent and fearless in combat and wouldn't shy away from fighting out numbered. Viking Combat works with axes, spears, and swords focuses on group combat. Below is a buffs and techniques provided by Viking Combat:

OnBlock Buff: "Battle Ready" - +1 Dodge, Stacks 4 times, duration 2
OnDodge Buff: "Fearless" - +1 Block, Stacks 4 times. duration 2
OnHit Buff: "Determination" - +1 bash, cut, stab armor penetration. Stack 20 times, duration 2
OnMove Buff: "Surprise Attack" - Increases all damage equal to 10% of Strength. Stack 5 times, duration 2
Technique: "Fake Out" - Needs Melee 1. Feint / Miss recovery
Technique: "Sweeping Strike" - Needs Melee 3. Arc AoE, weighting 2
Technique: "Whirlwind" - Needs Melee 4. Crit only, Circle AoE

Note: AoE techniques CANNOT be triggered against single enemies. They are coded to only trigger against 2 or more enemies that are adjacent to the player. 

A new martial arts book for Viking Combat called "Warriors of the North" has been added to the game. It can rarely be found in mansions like the rest of the melee weapon martial arts.

The Melee Weapon Training trait has also been updated to include Viking Combat as well.
#### Describe alternatives you've considered
<!--
A clear and concise description of any alternative solutions or features you've considered.
-->
Balancing this martial art is extremely difficult. Originally, it was based of the Berserker Viking legend. However, it proved too hard to balanced and also had a problem with realism. The lack of shields also hinders the style but there is nothing I can do about that. The martial art is group focused but uses short duration, stacking buffs to encourage players to tough it out in melee for as long as possible.
#### RFC

I worked hard to balance this martial art and make it interesting. But I am always open to feedback on how to make it better or more balanced.

KITbutler commented:
I see nothing preventing the use of this as a straight-up damage buff in 1v1 tactical nightcrawling. At the very least you should try to find a way of adding large amounts of noise to every technique including the miss recovery. The "static" buff should be moved to "on being hit", "on block" or "on dodge" with a very short duration, weak effect but high stack count so it favours 1vMANY over 1v1. 

Reduced movecost is almost always better than raw damage due to the way the move economy works (you regain your dodges on every one of *your* turns not a fixed 100moves timer). Those should definitely be removed from the AoE techniques and having rapid strike alone is a 30% dpm increase which is iffy as well.

The broadsword, cavalry sabre, cutlass, nodachi, zweihander and scimitar are used very differently from Viking-era swords and Dane axes and would not be usable to their full extent. The jian, katana, longsword and wakizashi (which is listed twice btw) are ahistorical and/or from a different culture and would not be immediately usable either, although they might be close enough in some regards to use the basic techniques.

And then of course the elephant in the room: Viking Berserkers were not real. Or rather the position *did* exist but they were champions in duels rather than naked ale-fueled killing machines. I would move the entire thing to the "Norse weapons" mod as it is far more in line with attempting to replicate Mjölnir than historical Viking-era combat (which was mostly fought using spears, large round shields and *a few* two-handed axes with swords and one-handed axes as sidearms)

ifreund commented:
>Viking Combat works with all axes and swords including short swords

I'm not sure it's appropriate to include the single edged japanese swords in a viking martial style.  They already have their own style and they certainly weren't historically used by vikings.

>Technique: "Sweeping Strike" - Needs Melee 3. Arc AoE, weighting 2, 50% move cost
Technique: "Whirlwind" - Needs Melee 4. Crit only, Circle AoE, 80% move cost

These move costs seem very low for attacks that can potentially hit multiple enemies. 

More general comments:
It's worth noting that this doesn't include spears which were historically the most common weapon used by vikings.   The round shield, which was incredibly important to viking combat, is also missing though that's of course understandable as there are as of yet no shields in the game.  I guess my main issue with this PR is that this doesn't seem to make much of an attempt to replicate historical viking combat, rather how popular culture portrays vikings.

Mecares commented:
Cool Idea, but I am not sure it fits into the base game really well, maybe it would fit better into the medieval mod?

Things I would associate with Viking combat style:
Bonus to throwing axes
Bonus after yelling maybe even stackable to represent yourself pushing yourself into your Berserker Rage
Some Bonus while standing in a floating vehicle/boat
Bonus while under the influence of alcohol maybe additional pain reduction?

Eliijahh commented:
> > Technique: "Sweeping Strike" - Needs Melee 3. Arc AoE, weighting 2, 50% move cost
> > Technique: "Whirlwind" - Needs Melee 4. Crit only, Circle AoE, 80% move cost
> 
> These move costs seem very low for attacks that can potentially hit multiple enemies.

I would not mind if this moves were a bit stronger than other martial arts techniques with, perhaps, the caveat that you need to take a drug in order to have a chance to trigger them? This would make this technique a bit less magical and in-line to what it is believed the driving cause of going "Berserk".

tenmillimaster commented:
So, uh, where can I sign up for my local Viking combat classes? 


/S.

But seriously, how is the survivor supposed to learn this? There aren't exactly martial arts manuals published on this stuff, let alone modern practitioners.

Hymore246 commented:
I appreciate the feedback everyone. Let me address a few of the concerns brought up.

Weapons Choices
Fair point on these. I included those weapons mainly to cover as many sword and axe types as possible. I can remove a lot of them for realism. I'll try to narrow the scope and keep the technique "basic" so it wouldn't be too much of a stretch to image using a machete for example.

Spears
I excluded spears mainly due to Sojutsu. If I include them I need to make sure Viking Combat doesn't completely outclass Sojutsu when using a spear. 

Berserkers
Real or not, I can completely remove this aspect. It's there mainly because "Barbarian Rage" is a recognizable trope in video games and got the point across about what the martial art was about. That being, walking into a group and slaughtering them with a sword or an axe.

1v1, 1vMany
Other AoE styles like Drunken Boxing rely on counters to kill groups. This is fine I guess but I wanted Viking Combat to be more offensive. Outside of weapon specific techniques and Bionic Combat's Cleave, I don't think any martial uses offensive AoE techniques. This is sort of new territory when doing balance. 

The only way to make this "purely" AoE Combat is to remove anything the works against a single target and I think that makes it too niche. I can remove the speed reductions and maybe Quick Hack in favor of a better buff that builds as you hit enemies. Let it have a short duration but can stack really high. That way, the AoE techniques can build it fast while not being worthless against single targets. I think the feint should stay. Missing in a group can be really bad. It's either that or boost accuracy somehow.

Learning Viking Combat
Go find your local longboat and sign up today! /S But, I always imaged that you could find a history book detailing how Vikings fought and use that as a basis for the martial art. If I remove the berserker aspect, this makes even more sense.

Viking Theme
At it's core, this is a martial art that uses swords and axes and can fight groups. Vikings seemed like the most likely group that would do this that wasn't already represented in the game. I could change it to something else if someone can make a good case for it.

tenmillimaster commented:
This sounds like good mod content, but not very realistic as there aren't any modern practitioners. You'd have to recreate the style, rather than learn a known style.

Now, if you were proposing HEMA as a style, I could get behind that as there are indeed modern practictioners. Consider reworking the request with that in mind, maybe?

KITbutler commented:
> But, I always imaged that you could find a history book detailing how Vikings fought and use that as a basis for the martial art.

The issue here is that the Vikings simply did not fight like anything resembling your gameplay vision. The closest you can find from that period would probably be a powercrept and romanticised [Housecarl](https://en.wikipedia.org/wiki/Housecarl) specifically with a Dane axe. The closest you could get with any other historical warrior would probably be Renaissance greatsword/Montante as these were used to charge headfirst into dense infantry formation and break them up (wearing full plate armour and mostly combating pikes though)

> Now, if you were proposing HEMA as a style, I could get behind that as there are indeed modern practictioners. Consider reworking the request with that in mind, maybe?

The Medieval mod has Medieval Swordsmanship which to me as a casual non-practitioner sadly bears little resemblance to HEMA. Granted since most intermediate/advanced HEMA training seems to be about hitting things **without getting hit yourself** an accurate representation would be ridiculously OP against unarmed zombies.

tenmillimaster commented:
> Granted since most intermediate/advanced HEMA training seems to be about hitting things without getting hit yourself an accurate representation would be ridiculously OP against unarmed zombies.

That's, er, what a lot of martial arts are about. Some reasonable approximation could be had.

Hymore246 commented:
Alright, rebalanced the martial to be based on historic Viking Warriors instead of the Berserker legend.
I used these as references:
https://en.wikipedia.org/wiki/Viking_raid_warfare_and_tactics
https://en.wikipedia.org/wiki/Viking_Age_arms_and_armour

Viking Combat looks like this now:
OnBlock Buff: "Battle Ready" - +1 Dodge, Stacks 4 times
OnDodge Buff: "Fearless" - +1 Block, Stacks 4 times.
OnHit Buff: "Determination" - +1 bash, cut, stab armor penetration. Stack 20 times
OnMove Buff: "Surprise Attack" - Increases all damage equal to 10% of Strength. Stack 5 times
Technique: "Fake Out" - Needs Melee 1. Feint / Miss recovery
Technique: "Sweeping Strike" - Needs Melee 3. Arc AoE, weighting 2
Technique: "Whirlwind" - Needs Melee 4. Crit only, Circle AoE

Realism
Real Vikings were violent and fearless. They liked to use surprise attacks against places with little defense. They would attack in a wedge formation but would also use hit and run tactics if possible. Nothing indicates that they were louder than anyone else when fighting. The lack of armor wasn't a preference, they were just poor. Only the richest Vikings had chain armor and helmets. I removed all non-cultural weapons and added spears as well to be more historically accurate.

The martial art is more condensed and I think it actually works better now.

KITbutler commented:
I think the cumulative boosts are still way too high. Reduced movecost on AoE attacks is ridiculous without external drawbacks (i.e. it would probably be okay if it only worked with a bullwhip). I would scrap the onmove buff because it's a straight-up damage boost (can you make arpen scale with STR instead?) 

In general you probably need to rethink the core theme, do you want a mobile medieval warrior or do you want a killing machine who wades into a horde and stomps out on the other side covered in zombie guts?

Hymore246 commented:
Whoops, those move cost reductions should have been removed. I can get rid of the move buff and even the onhit buff and just give straight strength to arpen as a static buff. 

I want this to be an AoE style but I also don't want to to be redundant. There isn't a point to adding it if it is automatically out classed in AoE by other martial arts like Drunken Boxing or anything with a counterattack.

Based on your suggestions, this it what Viking Combat would look like:
Static Buff: "Determination" -  Bash, cut, and stab armor penetration equal to 100% of Strength
OnBlock Buff: "Battle Ready" - +1 Dodge, Stacks 4 times, duration 2
OnDodge Buff: "Fearless" - +1 Block, Stacks 4 times. duration 2
Technique: "Fake Out" - Needs Melee 1. Feint / Miss recovery
Technique: "Sweeping Strike" - Needs Melee 3. Arc AoE, weighting 2
Technique: "Whirlwind" - Needs Melee 4. Crit only, Circle AoE

Does this sound more reasonable?

Hymore246 commented:
I thought Zui Quan was nerfed a while back. It's still broken? I'm not following your train of thought on the arpen. I think it's fine to build arpen against anything, group or single target. Most of the time you will not make it to the max even if I limited it to max strength. If you do, you are either fighting a horde or doing next to no damage to something like a hulk or juggernaut.

What do you mean by zero cost AoEs? The cost the same any other attack. Aside from Sweep being non crit, they are the same as the weapon technique versions.

KITbutler commented:
Zui Quan **was** nerfed and I have not played with it enough to have a well-informed opinion on the current balance, but *you* stated concerns about it still being OP so I gave you the proper procedure for addressing that issue.

The problem with static buffs is that you get the full bonus even when running into a lone enemy at night. Requiring you actually *hit* things ensures the buff is not active at the start of a fight and falls off quickly once enemies are dealt with, meaning you gain almost no benefit if you draw them out one by one and remain vulnerable to surprise attacks.

With "zero cost" I mean you are not giving up any of your *existing* capabilities to gain the AoE capability, especially since there are few competing weapon MAs. The existing AoE weapons have several undesirable qualities such as being large, heavy, slow and potentially lacking other techniques such as reach attacks or rapid strike.

Hymore246 commented:
After chatting with the devs, I've decided develop this as a mod instead of adding it straight to the game. I'll continue to refine the martial art in the meantime.
